app/services/nomis/client.rb

```diff
@@ -64,5 +87,20 @@ def params_options(method, params)
         #   }
       end
     end
+
+    def auth_header
+      return unless @client_token && @client_key
+
+      token = auth_token(@client_token, @client_key)
```

Comment:
Definitely a reasonable point, however I prefer to make it explicit that `auth_token` is a function of the client token and key. Does that make sense?